content/docs/05.concepts/02.namespace-files.md

```diff
@@ -304,10 +304,85 @@ The example above uses the `include` field to only allow the `scripts/weather.py
 
 We can control what namespace files are available to our flow with the `namespaceFiles` property.
 
-`namespaceFiles` has 3 attributes:
+`namespaceFiles` has several configurable attributes:
 - `enabled`: when set to true enables all files in that namespace to be visible to the task
 - `include`: specifies files you want to be accessible by the task
 - `exclude`: specifies files you don't want to be accessible by the task
+- `namespaces`: specifies a list of namespaces to search for files.
+- `ifExists`: specifies what to do in the instance a Namespace file already exists in the working directory
+
+The `namespaces` attribute can be used like in the following example:
+
+```yaml
+id: hyena_375528
+namespace: dev.test
+
+tasks:
+
+  - id: namespace
+    type: io.kestra.plugin.scripts.python.Commands
+    namespaceFiles:
+      enabled: true
+      namespaces:
+        - "dev.test"
+        - "company"
+    commands:
+      - python test.py
+
+
+  - id: namespace2
+    type: io.kestra.plugin.scripts.python.Script
+    namespaceFiles:
+      enabled: true
+      namespaces:
+        - "dev.test"
+        - "company"
+    script: "{{ read('test.py') }}"
+```
+
+The files are loaded in the namespace order, and only the latest version of a file is kept. Meaning if a file is present in the first and second namespace, only the file present on the second namespace will be loaded. In the first task, the `test.py` file from the `company` namespace will be used because priority is given from top to bottom of the list of namespaces. In the case of multiple files of the same name, the last listed namespace holds priority.
+
+For the second task, the `test.py` file in the `dev.test` namespace will be used. When using the `read()`, the priority is changed from the ordering to whatever the current namespace in the task is running, in this case `dev.test`.
+
+Lastly, you can specify in the `read()` function which namespace to use such as `"{{ read('test.py', namespace='company') }}"`. This ensures priority is given to whatever you specify without having to consider ordering or current namespace.
```

Comment:
```suggestion
```

it's not priority of namespace loading, but explicit rendering of function arguments, best to just cut it